src/components/activity-completion/completion-export-answers.tsx

```diff
@@ -0,0 +1,35 @@
+import React from "react";
+import { Storage } from "../../storage-facade";
+
+import "./completion-export-answers.scss";
+
+interface IProps { }
+interface IState {
+  activityJSON: string
+}
+
+export class CompletionExportAnswers extends React.PureComponent<IProps, IState> {
```

Comment:
`React.PureComponent` should work here since the state is "shallow" (only has top level keys).  But it should probably be changed to `React.Component` in case the state is changed to have "deep" keys as it will then subtly break as noted in the React docs:

> React.PureComponent’s shouldComponentUpdate() only shallowly compares the objects. If these contain complex data structures, it may produce false-negatives for deeper differences. Only extend PureComponent when you expect to have simple props and state, or use forceUpdate() when you know deep data structures have changed. Or, consider using immutable objects to facilitate fast comparisons of nested data.

> Furthermore, React.PureComponent’s shouldComponentUpdate() skips prop updates for the whole component subtree. Make sure all the children components are also “pure”.

Reply:
if we change the link to build the json on click, similar to the offline manifest builder, then I don't think this will be an issue? I'm not sure this button will do much else